plugin/src/main/java/org/opensearch/sql/plugin/rest/RestPPLQueryAction.java

```diff
@@ -100,10 +100,11 @@ public void onResponse(TransportPPLQueryResponse response) {
               public void onFailure(Exception e) {
                 if (transportPPLQueryRequest.isExplainRequest()) {
                   LOG.error("Error happened during explain", e);
-                  sendResponse(
-                      channel,
-                      INTERNAL_SERVER_ERROR,
-                      "Failed to explain the query due to error: " + e.getMessage());
+                  if (isClientError(e)) {
+                    reportError(channel, e, BAD_REQUEST);
+                  } else {
+                    reportError(channel, e, INTERNAL_SERVER_ERROR);
+                  }
```

Comment:
FYI: This change is fixing an existing bug which is unable to display the explain failure.